Add bin and asset dir to gitignore

hulsiejames opened the pull request:
Adds the bin and assets directories to .gitignore

Should help clear up the changes files tracker when working on updates. 

The bin & assets dir should always be empty when the repo is cloned. These directories are populated with setup.py (bin), and by the user (assets)

If this is not a useful edit I can remove - thoughts appreciated

wsp-mbuckley commented:
This shouldn't be necessary because both of those folders already contain `.gitignore` files which ignore everything else within the folder. Have you ran into an issue with accidentally commiting files in either folder?

wsp-mbuckley commented:
Although, it could be worth removing the assets folder altogether and switching to just having full paths in the configs instead of searching in the assets folder?

hulsiejames commented:
> This shouldn't be necessary because both of those folders already contain `.gitignore` files which ignore everything else within the folder. Have you ran into an issue with accidentally commiting files in either folder?

You're right! 

If I'm being honest I'm not too sure what was going on my end. Whenever I have added files to the assets or bin directory these have been appearing in my tracked files.

I've just cleared out my files as these were up to date and re-cloned the repo and is now working as expected!

(I've added a test.xt file to assets dir and it is no longer tracking this)
![git tracking](https://github.com/Transport-for-the-North/otp4gb-py/assets/58815827/ffae3189-64e4-43bb-9e8e-920283bce8c0)

I hadn't comitted anything by mistake so far, it was just a small headache trying to find the files I wanted to commit among 70 changes files.

I'll delete this PR I think

hulsiejames commented:
> Although, it could be worth removing the assets folder altogether and switching to just having full paths in the configs instead of searching in the assets folder?

I think it works well having asset folder for now. A single place a user can put all required data and it keeps the config looking a bit cleaner having filenames rather than full paths. 

I can see the config becoming a bit chaotic with the addition of full paths, especially if some data are nested in multiple directories. Though happy to hear your thoughts on this.
